cuda/solver/batch_bicgstab_kernels.cu

```diff
@@ -56,19 +75,217 @@ namespace cuda {
 namespace batch_bicgstab {
 
 
+#include "common/cuda_hip/base/batch_multi_vector_kernels.hpp.inc"
+#include "common/cuda_hip/components/uninitialized_array.hpp.inc"
+#include "common/cuda_hip/matrix/batch_dense_kernels.hpp.inc"
+#include "common/cuda_hip/matrix/batch_ell_kernels.hpp.inc"
+#include "common/cuda_hip/solver/batch_bicgstab_kernels.hpp.inc"
+
+
+template <typename StopType, typename PrecType, typename LogType,
+          typename BatchMatrixType, typename ValueType>
+int get_num_threads_per_block(std::shared_ptr<const DefaultExecutor> exec,
+                              const int num_rows)
+{
+    int num_warps = std::max(num_rows / 4, 2);
+    constexpr int warp_sz = static_cast<int>(config::warp_size);
```

Comment:
from this, it seems to use a warp to handle a row.
but the following check is using a thread to handle a row.

Reply:
I think the strategy (warp to row or thread to row) can depend on the matrix format. Hence, this just ensures that a minimum number of warps is always used.